File: API/AmusementPark.WebAPI/Contracts/Videos/ResolvedVideoMetadataDto.cs
namespace AmusementPark.WebAPI.Contracts.Videos;

public sealed class ResolvedVideoMetadataDto
{
    public VideoHostingProviderDto HostingProvider { get; set; }

    public string OriginalUrl { get; set; } = string.Empty;

    public string CanonicalUrl { get; set; } = string.Empty;

    public string? EmbedUrl { get; set; }

    public string? ExternalId { get; set; }

    public string? Title { get; set; }

    public string? Description { get; set; }

    public string? CreatorName { get; set; }

    public string? CreatorUrl { get; set; }

    public string? ThumbnailUrl { get; set; }

    public long? DurationSeconds { get; set; }

    public DateTime? PublishedAtUtc { get; set; }

    public string? MetadataSource { get; set; }

    public DateTime? FetchedAtUtc { get; set; }

    public string? ProviderChannelId { get; set; }

    public string? ProviderChannelUrl { get; set; }
}

File: FRONT/AmusementPark/src/app/models/videos/resolved-video-metadata-dto.ts
import { VideoHostingProvider } from './video-hosting-provider';

export interface ResolvedVideoMetadataDto {
  hostingProvider: VideoHostingProvider;
  originalUrl: string;
  canonicalUrl: string;
  embedUrl?: string | null;
  externalId?: string | null;
  title?: string | null;
  description?: string | null;
  creatorName?: string | null;
  creatorUrl?: string | null;
  thumbnailUrl?: string | null;
  durationSeconds?: number | null;
  publishedAtUtc?: string | null;
  detectedLanguageCode?: string | null;
  metadataSource?: string | null;
  fetchedAtUtc?: string | null;
  providerChannelId?: string | null;
  providerChannelUrl?: string | null;
}
